### ISSUE TITLE
[suggestion] Let list methods take an Option argument

### ISSUE THREAD

**Issue by vrurg** [64bit/async-openai#243]
Maybe this is my lack of experience, but when I'm requesting just a full list of objects like files I have to write something like:

```rust
    let file_list = files.list::<[&str; 0]>(&[]).await?;
```

Wouldn't it be better to let it be just  `files.list(None).await?`?

**Comment by 64bit** [comment]
Thanks for suggestion, it can be improved, how about having two functions 
1. `files.list()` with no parameters, which calls  another public function below
2. `files.list_with_query(query: Q)` 

I think `files.list()` has better ergonomics than `files.list(None)`.

**Comment by vrurg** [comment]
> I think `files.list()` has better ergonomics than `files.list(None)`.

I agree. I just followed #167 and tried to figure out a good compromise.

BTW, a side note here. The query arguments are currently undocumented (vector store `list` method uses the same design). Perhaps it worth pointing out at `reqwest` crate for those looking for understanding without digging into the sources, like I did.

**Comment by 64bit** [comment]
That's a good observation too about lack of documentation on `query` parameter - any contribution to help improve it is most welcome!   I think one of examples have a usage but that's not very visible/searchable on docs.rs)

Discussion in #167 was in context of missing the call for Files api group. It does seem like it needs to be addressed across the whole crate, that does sound like a breaking change where existing `list` becomes `list_with_query` 

Its a small change and and an improvement so why not? Can be released in a version bump.

**Comment by vrurg** [comment]
> That's a good observation too about lack of documentation on `query` parameter - any contribution to help improve it is most welcome! (I think one of examples have a usage but that's not very visible/searchable on docs.rs)

Oh, I know this tone... :) Unfortunately, as you can see, I barely find time on side-projects now. Had to even quit Raku development. Otherwise it'd be my pleasure!

**Comment by 64bit** [comment]
No obligations, thank you for your contributions, the issue is going to be here if you wanna give it a shot.
